### PR TITLE
USB: Gametrak/RealPlay emulation

### PR DESCRIPTION
**Pull request by Florin9doi**
### Description of Changes
Bring Gametrak/RealPlay controller emulation from wxWidgets (#5032) to QT.
This time with new options : axis inversion and axis limitation which should increase compatibility with standard gamepads (for example in RealPlay Racing), although the original controllers are still necessary for higher compatibility.

### Notes
- Most of the time only half-axis are captured in settings. It may be necessary to edit `PCSX2.ini` and replace `SDL-#/±Axis#` with `SDL-#/FullAxis#` : `SDL-0/FullAxis0`, `SDL-0/FullAxis1`, ... 

### Suggested Testing Steps
Check some of the following games:
- Gametrak
  - [Real World Golf](https://wiki.pcsx2.net/Real_World_Golf)
  - [Real World Golf 2007](https://wiki.pcsx2.net/Real_World_Golf_2007)
  - [Dark Wind](https://wiki.pcsx2.net/Dark_Wind)
- RealPlay
  - [RealPlay Golf](https://wiki.pcsx2.net/RealPlay_Golf)
  - [RealPlay Racing](https://wiki.pcsx2.net/RealPlay_Racing)
  - [RealPlay Puzzlesphere](https://wiki.pcsx2.net/RealPlay_Puzzlesphere)
  - [RealPlay Pool](https://wiki.pcsx2.net/RealPlay_Pool)

### CODE REVIEW

**Review by lightningterror — COMMENTED** [review]
A couple of new compiler warnings creeped up that can be fixed:
https://github.com/PCSX2/pcsx2/actions/runs/10060812620/job/27809310067?pr=11617#step:14:745
https://github.com/PCSX2/pcsx2/actions/runs/10060812620/job/27809310067?pr=11617#step:14:976

**Comment by Florin9doi** [comment]
fixed

File: pcsx2/USB/usb-pad/usb-realplay.cpp [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+
+	fail:
+		realplay_unrealize(&s->dev);
+		return nullptr;
+	}
+
+	bool RealPlayDevice::Freeze(USBDevice* dev, StateWrapper& sw) const
+	{
+		RealPlayState* s = USB_CONTAINER_OF(dev, RealPlayState, dev);
+
```
Comment (F0bes) [inline pcsx2/USB/usb-pad/usb-realplay.cpp:347]: Although not shown in GitHubs diff tool for some reason, there appears to be 2 tabs on this blank line.
If you check [here](https://github.com/PCSX2/pcsx2/raw/76de27392197da01d91257be1e7f93609e711b55/pcsx2/USB/usb-pad/usb-realplay.cpp) it's present.

File: pcsx2/USB/usb-pad/usb-gametrak.cpp [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+	fail:
+		gametrak_unrealize(&s->dev);
+		return nullptr;
+	}
+
+	bool GametrakDevice::Freeze(USBDevice* dev, StateWrapper& sw) const
+	{
+		GametrakState* s = USB_CONTAINER_OF(dev, GametrakState, dev);
+
```
Comment (F0bes) [inline pcsx2/USB/usb-pad/usb-gametrak.cpp:318]: Same issue here. 2 tabs on this line.

**Review by F0bes — CHANGES_REQUESTED** [review]
A couple issue with some tabs.
I don't have a game I can use to thouroughly test it, code appears fine, I trust you have it working.

File: pcsx2/USB/usb-pad/usb-realplay.cpp (hunk 1 above)
Comment (Florin9doi) [reply pcsx2/USB/usb-pad/usb-realplay.cpp]: fixed

File: pcsx2/USB/usb-pad/usb-gametrak.cpp (hunk 2 above)
Comment (Florin9doi) [reply pcsx2/USB/usb-pad/usb-gametrak.cpp]: fixed

**Review by F0bes — APPROVED** [review]
Thanks, LGTM

**Pull request merged**